django/db/utils.py

```diff
@@ -197,6 +199,89 @@ def create_connection(self, alias):
         return backend.DatabaseWrapper(db, alias)
 
 
+class AsyncAlias:
+    """
+    A Context-aware list of connections.
+    """
+
+    def __init__(self) -> None:
+        self._connections = Local()
+        setattr(self._connections, "_stack", [])
+
+    @property
+    def connections(self):
+        return getattr(self._connections, "_stack", [])
+
+    def __len__(self):
+        return len(self.connections)
+
+    def __iter__(self):
+        return iter(self.connections)
+
+    def __str__(self):
+        return ", ".join([str(id(conn)) for conn in self.connections])
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}: {len(self.connections)} connections>"
+
+    def add_connection(self, connection):
+        setattr(self._connections, "_stack", self.connections + [connection])
+
+    def pop(self):
+        conns = self.connections
+        conns.pop()
+        setattr(self._connections, "_stack", conns)
+
+
+class AsyncConnectionHandler:
```

Comment:
Here are a few thoughts about that

- Django has `BaseConnectionHandler`, which provides a base class for managing connections (cache/database etc). We can follow this way and implement `BaseAsyncConnectionHandler` and reuse it for cache connection as well
- I think keeping the same interface for connection and asynchronous connection is important. It increases the amount of code that we can reuse and, as a result, decrease code duplication. For these purposes, we can organize all logic related to the stack into a `StackLocal` instance 

example of the implementation - https://github.com/Arfey/django/pull/1/commits/be7f02cc18af0237c8a00663ae8c8a0ee0c4d3f2

what do u think?

Reply:
@fcurella if it looks good for u I can help and prepare MR for your branch